.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -418,12 +426,26 @@ private DateTimeResolutionResult ParseDateWithTimePeriodSuffix(string text, Date
             if (dateEr != null && timeEr != null)
             {
                 var dateStrEnd = (int)(dateEr.Start + dateEr.Length);
+                var timeStrEnd = (int)(timeEr.Start + timeEr.Length);
 
-                if (dateStrEnd < timeEr.Start)
+                // if CheckBothBeforeAfter, it checks also for patterns where time precedes date
+                if (dateStrEnd < timeEr.Start || (this.Config.CheckBothBeforeAfter && timeStrEnd < dateEr.Start))
                 {
-                    var midStr = text.Substring(dateStrEnd, timeEr.Start.Value - dateStrEnd).Trim();
+                    var midStr = dateStrEnd < timeEr.Start ? text.Substring(dateStrEnd, timeEr.Start.Value - dateStrEnd).Trim() :
+                        text.Substring(timeStrEnd, dateEr.Start.Value - timeStrEnd).Trim();
+                    var afterStr = text.Substring(timeStrEnd);
+
+                    bool inPrefix = true;
+                    string modStr = GetValidConnectorModForDateAndTimePeriod(midStr, inPrefix);
```

Comment:
In such cases, it's more readable to use inline named arguments, like:
`string modStr = GetValidConnectorModForDateAndTimePeriod(midStr, inPrefix: true);`

Reply:
noted, we will update the file

Reply:
Can you also check other recently changed files for the same for consistency? Thanks.